### DISCUSSION TITLE
Export AdGuard Home Configuration

### DISCUSSION THREAD

**Discussion by elipticon2** [AdguardTeam/AdGuardHome#4032] [Ideas]
It would be much helpful would it be to be able to export the following configurations:

1. Client List
2. DNS Upstream servers list
3. Custom Filtering Rules
4. Blocked services
5. DNS Blocklists
6. DNS Allow Lists
7. Query and Statistics Retention Period

This would help in scaling up the configuration as there is no clustering mechanism with it. This would make my life very very easy.

**Comment by ainar-g** [comment]
Hello.  All of these are already kept in the configuration file, `AdGuardHome.yaml`.  For actual import and export HTTP APIs, there are feature requests #1147 and #573.

**Reply by elipticon2** [reply]
Thanks a lot for the great help!
